test/extended/openstack/cpms.go

```diff
@@ -81,45 +91,87 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] ControlPlane Mach
 			machineFlavor := machine.Get("spec.providerSpec.value.flavor").String()
 			machineImage := machine.Get("spec.providerSpec.value.image").String()
 			machineSecurityGroups := make(map[string]struct{})
 			machineSubnets := make(map[string]struct{})
-			machineServerGroup := machine.Get("spec.providerSpec.value.serverGroupName").String()
-			machineNovaAvailabilityZone := machine.Get("spec.providerSpec.value.availabilityZone").String()
-
-			for _, sg := range objects(machine.Get("spec.providerSpec.value.securityGroups")) {
-				securityGroupName := sg.Get("name").String()
+			machineServerGroup := machineProviderSpec.Get("serverGroupName").String()
+			machineNovaAz := machine.Get("spec.providerSpec.value.availabilityZone").String()
+			machineCinderAz := machine.Get("spec.providerSpec.value.rootVolume.availabilityZone").String()
+			machineVolumeType := machine.Get("spec.providerSpec.value.rootVolume.volumeType").String()
+			var machineNetworks []string
+
+			for _, sg := range objects(machineProviderSpec.Get("securityGroups")) {
+				securityGroupName := sg["name"].(string)
 				machineSecurityGroups[securityGroupName] = struct{}{}
 			}
 
 			for _, subnet := range objects(machine.Get("spec.providerSpec.value.networks[0].subnets")) {
 				subnetName := subnet.Get("filter.name").String()
 				machineSubnets[subnetName] = struct{}{}
 			}
-			if machineNovaAvailabilityZone != "" {
-				machineNovaAvailabilityZones = append(machineNovaAvailabilityZones, machineNovaAvailabilityZone)
+
+			if machineNovaAz != "" {
+				machineFd["nova_az"] = machineNovaAz
+			}
+
+			if machineCinderAz != "" {
+				machineFd["cinder_az"] = machineCinderAz
+			}
+			if machineVolumeType != "" {
+				machineFd["volume_type"] = machineVolumeType
 			}
+			if len(machineFd) != 0 {
+				machinesFds = append(machinesFds, machineFd)
+			}
+
+			for _, network := range objects(machine.Get("spec.providerSpec.value.networks")) {
+				subnet := network.Get("subnets").String()
+
+				if subnet == "" {
+					machineNetworks = append(machineNetworks, network["uuid"].(string))
+				}
+			}
+
 			o.Expect(machineFlavor).To(o.Equal(cpmsFlavor), "flavor mismatch on Machine %q", machineName)
 			o.Expect(machineImage).To(o.Equal(cpmsImage), "image mismatch on Machine %q", machineName)
 			o.Expect(machineSecurityGroups).To(o.Equal(cpmsSecurityGroups), "security group mismatch on Machine %q", machineName)
 			o.Expect(machineSubnets).To(o.Equal(cpmsSubnets), "subnets mismatch on Machine %q", machineName)
 			o.Expect(machineServerGroup).To(o.Equal(cpmsServerGroup), "server group	 mismatch on Machine %q", machineName)
+			o.Expect(machineNetworks).To(o.Equal(cpmsNetworks), "Network mismatch on Machine %q", machineName)
 		}
-		sort.Strings(machineNovaAvailabilityZones)
-		o.Expect(machineNovaAvailabilityZones).To(o.Equal(cpmsNovaAvailabilityZones))
+		o.Expect(sortFd(machinesFds)).To(o.Equal(sortFd(cpmsFds)))
```

Comment:
Luckily, Gomega [has a matcher](https://github.com/itzikb-redhat/openstack-test/blob/18c279561a659ede4a11afc53ce5f4b25465a151/vendor/github.com/onsi/gomega/matchers.go#L349) that doesn't care about ordering. You should be able to drop the sorting logic:

```suggestion
		o.Expect(machinesFDs).To(o.ConsistOf(cpmsFDs))
```